lib/assets/javascripts/turbolinks.js.coffee

```diff
@@ -177,12 +177,10 @@ swapNodes = (targetBody, existingNodes, options) ->
       else
         targetNode = targetNode.cloneNode(true)
         existingNode.parentNode.replaceChild(targetNode, existingNode)
-        if targetNode.nodeName == 'SCRIPT' && targetNode.getAttribute("data-turbolinks-eval") != "false"
-          executeScriptTag(targetNode)
```

Comment:
@kristianpd what was the use-case for this?

Right now in `master`:
- `<script data-turbolinks-temporary>` is eval-ed twice on full replace (second time [here](https://github.com/rails/turbolinks/blob/d12cc831e113b0299ec2090d8ecfe6ff5f2f9a53/lib/assets/javascripts/turbolinks.js.coffee#L151))
- `<script data-turbolinks-temporary>` is eval-ed once on partial replace
- `<script id="change">` is eval-ed once on partial replace with `change: ['change']`
- `<script id="change" data-turbolinks-temporary>` is eval-ed twice on partial replace with `change: ['change']`

In this PR:
- `<script data-turbolinks-temporary>` is eval-ed once on full replace
- `<script data-turbolinks-temporary>` is not eval-ed on partial replace
- `<script id="change">` is not eval-ed on partial replace with `change: ['change']`
- `<script id="change" data-turbolinks-temporary>` is not eval-ed on partial replace with `change: ['change']`

I'm guessing we want:

- `<script data-turbolinks-temporary>` is eval-ed once on full replace
- `<script data-turbolinks-temporary>` is eval-ed once on partial replace
- `<script id="change">` is eval-ed once on partial replace with `change: ['change']`
- `<script id="change" data-turbolinks-temporary>` is eval-ed once on partial replace with `change: ['change']`

?

cc @dhh